README.md
# Paris Living Style Guide

## Getting started

Install [Harp](http://harpjs.com):

    npm install -g harp

And Browserify

    npm install -g browserify watchify

And browser-sync (*)

    npm install -g browser-sync

(*) You may not need browser-sync, it can be a bit complicated to [install on Windows](http://www.browsersync.io/docs/#windows-users) and is only needed if you want your pages to auto-reload on code changes.

And the project dependencies (run this command from the project folder)

    npm install


## Development mode

    npm start

This launches a local server at http://localhost:9000
The server is proxied on http://localhost:3000 by browser-sync with the auto-reload on code changes.


## Compile a static version

    npm run compile


## Build the styleguide

    npm run build

This task will:
* compile all SCSS files into a single `./build/assets/css/paris.css`
* compile all JS files into a single `./build/assets/javascript/paris.js`
* copy `./src/javascript/config.js` to `./build/assets/javascript/config.js` without modification
* copy `./src/javascript/locales.js` to `./build/assets/javascript/locales.js` without modification
* copy all webfont files to `./build/assets/fonts/`
* optimize and copy all images (png, jpg, svg, gif) to `./build/assets/images/` while preserving the original tree structure
* copy all jade and json files to `./build/jade/` while preserving the original tree structure
* copy `./src/layouts/_wrapper.jade` to `./build/jade/layouts/wrapper.jade` without modification

### Use the build in another project

Run `npm link` in this folder

Run `npm link paris-styleguide` in the other project

You'll get the paris-styleguide project symbolically linked in the other project `node_modules`.

If it's an Express project you can do something like:

    app.set('views', [app.get('views'), path.join(__dirname, 'node_modules', 'paris-styleguide', 'build', 'jade') ]);

to be able to render Jade templates coming from the styleguide. \o/


CHANGELOG.md
# Changelog

## Versioning system

Let's take the `x.β.α` version number:
* `α` is for small fixes that don't involve any template update, mainly css evolution
* `β` is for versions that need template update for static (html) use of the styleguide and addition of new elements
* `x` is for major release involving a more careful integration of the styleguide

## x.β.α (coming soon)

## 1.4.4 (2016-08-24)

* accessibility: use `rem` instead of `px` for font units

## 1.4.3 (2016-07-20)

* module `sections-list`: improved rendering

## 1.4.2 (2016-07-19)

* fix `icons` (broken by 1.4.1)

## 1.4.1 (2016-07-19)

* `icons`: updated Sass syntax to use mixin instead of @extend (for compatibility)
* module `sections-list`: add colors and fix alignment

## 1.4.0 (2016-06-15)

* added module `block-content-tweet`

## 1.3.12 (2016-05-20)

* module `quick-access`: allow onglet:Professionnels from search

## 1.3.11 (2016-05-12)

* module `quick-access`: remove onglet:Professionnels from search

## 1.3.10 (2016-04-21)

* script `3975.js`: qualification navigation 3975

## 1.3.9 (2016-03-14)

* module `notice`: add grey background and fix center
* module `block-content-jecoute`: added pause button
* module `quick-access`: added pause button for background video
* module `poll`: added pause button for background animation

## 1.3.8 (2016-01-29)

* component `form`: add &nbsp; in cgu label

## 1.3.7 (2016-01-28)

* component `jecoute`: display previous error

## 1.3.6 (2016-01-26)

* component `jecoute`: handle API error

## 1.3.5 (2016-01-25)

* component `form`: better submitted data format
* module `pagination`: fixed CSS bug on small screens

## 1.3.4 (2016-01-18)

* new component `frame`
* component `text`: remove default `text` content

## 1.3.3 (2016-01-12)

* component `form`: add `citoyenne` input
* component `gallery`: remove dots

## 1.3.2 (2015-12-14)

* new component `districts`
* component `text`: change `h3` font-size to 27px
* template `generic`: added `heading_credit` as another `last-update` module

## 1.3.1 (2015-12-09)

* use new git worflow: see `CONTRIBUTING.md`
* module `button`:
  * added stateful button
  * allow to disable `<a>` tag buttons
  * more detailed documentation
* module `form`: style `cgu` link
* template `list-person`: fix js error on pagination rendering
* template `subhome-news`: allow to deactivate some modules

## 1.3.0 (2015-11-27)

* all components now accept `modifiers`
* component `card` added
* component `rush-hours` added
* module `block-aside-access` added
* module `block-aside-contact`: added minimap
* module `button`: add button `small`
* module `icon-switch` added
* module `notice`: improved text wrapping on small screens
* layout `left-col`: fix affix bug when there is no `anchors-list`
* template `single-equipment` added (based on `generic`)
* template `faq` is now based on `generic`
* template `generic`: `aside` data is now an array (same behavior as `components`)
* accessibility: improved modules `document-heading`, `rheader`, `sections-panel`
* DEPRECATED: removed favorite icon from `document-heading` (affects all templates based on `generic`)
* DEPRECATED: removed `share` from anchors (in component `text`)

## 1.2.2 (2015-11-24)

* component `form`: added `noscript` reCaptcha
* template `generic`: now accepts a `notice` above content
* upgraded harp to 0.20

## 1.2.1 (2015-11-19)

* do not cache ajax `post-it` request
* module `block-content-jecoute`: remove margin on `p` tag in answers

## 1.2.0 (2015-11-10)

* new UI for the styleguide documentation
* module `get-involved`: removed `title` from item
* component `video`: added `title`

## 1.1.6 (2015-11-06)

* accessibility improvements (mainly on `sections-panel` module)
* upgraded harp to 0.19

## 1.1.5 (2015-11-04)

* module `get-involved-list`: fixed collision with get-involved
* module `jumbotron`:
  * added `no-overlay` modifier to remove overlay
  * added conditional `block`

## 1.1.4 (2015-11-03)

* accessibility improvements
* component `jecoute` and module `poll`: use `button` module
* component `form` improved for release
* template `subhome-news`: bugs fixed in `news-list` and header
* improved Jade syntax in multiple files (always use `var`)

## 1.1.3 (2015-11-02)

* multiple accessibility improvements
* templates `document`, `single-news` and `single-person` have been merged into `generic`
* icon `.icon-file` renamed `.icon-dl` and added `.icon-file` `.icon-text`
* component `jecoute`: added `upload` and `policy` feature

## 1.1.2 (2015-10-02)

* same as 1.1.1, but with updated build

## 1.1.1 (2015-10-02)

* module `block-aside-checkboxes`: fix checkbox visibility on Chrome
* component `gallery`: add `clear: both`;

## 1.1.0 (2015-09-29)

* remove template jade from build
  * template js is still in the build
* component `form`: add missing .component class
* component `table`: style `a` tags as in text component
* template `list-person`:
  * fix js error on empty search
  * correct page number in url query

## 1.0.8 (2015-09-16)

* prod wrapper: allow `meta` and `link` tags injection
* update `templatizer` to 2.0.2
* component `form` added
* component `gallery` added
* component `postit`: added background color
* module `block-content-jecoute`: text can now contain links
* module `block-aside-checkboxes`: replace checkbox input by `a` tag for SEO
* template `list-persons`: improved search javascript

## 1.0.7 (2015-08-03)

* component `postit`: added `title` to data
* module `anchors-list`: add postit item dynamically
* module `gallery-ugc`: removed ajax loading
* fixed bug with `Cookies.getJSON` when the cookie is not set

## 1.0.6 (2015-07-27)

* component `jecoute`: fixed bug when after floating element
* component `news-push`: fixed link hover + optional link
* module `anchors-list` & `button-top`: a cookie can be used to disable scroll animation
* module `document-heading`: improved active & hover states for icons

examples/routes/index.js
var express = require('express');
var router = express.Router();


var data = {
	"data": {
		"notice": {
			"modifiers": ["top", "blue", "buildings"],
			"closable": true
		},
		"quick_access": {
			"background": "../images/modules/quick-access/background.jpg",
			"search": {
				"input": {
					"placeholder": "Rechercher un service, une information..."
				},
				"submit": {
					"text": "Rechercher"
				},
				"around": {
					"text": "Autour de moi"
				}
			},
			"items": [{
				"href": "#",
				"text": "Inscription à la crèche"
			}, {
				"href": "#",
				"text": "Cours municipaux pour adultes"
			}, {
				"href": "#",
				"text": "État civil"
			}, {
				"href": "#",
				"text": "Encombrants"
			}, {
				"href": "#",
				"text": "Tennis"
			}, {
				"href": "#",
				"text": "Signalement"
			}, {
				"href": "#",
				"text": "Facil’familles"
			}, {
				"href": "#",
				"text": "Gardes d’enfants"
			}],
			"notice": {
				"modifiers": ["bottom transparent"],
				"closable": true,
				"block": "En poursuivant votre navigation sur ce site, vous acceptez l’utilisation de Cookies pour vous proposer sed diam eget risus varius blandit sit amet non magna."
			}
		},
		"sections": {
			"heading": {
				"title": "Services et infos pratiques"
			},
			"list": {
				"items": [{
					"title": "Aides & démarches",
					"text": "Nullam id dolor id nibh ultricies vehicula ut id elit."
				}, {
					"title": "Culture & patrimoine",
					"text": "Sed posuere consectetur est at lobortis."
				}, {
					"title": "Déplacements & stationnement",
					"text": "Donec id elit non mi porta gravida at eget metus."
				}, {
					"title": "Emploi & formations",
					"text": "Sed posuere consectetur est at lobortis."
				}, {
					"title": "Environnement & espaces verts",
					"text": "Cras justo odio, dapibus ac facilisis in, egestas eget quam."
				}, {
					"title": "Famille & éducation",
					"text": "Donec ullamcorper nulla non metus auctor fringilla."
				}, {
					"title": "Innovation",
					"text": "Cras mattis consectetur purus sit amet fermentum."
				}, {
					"title": "Logement",
					"text": "Nullam quis risus eget urna mollis ornare vel eu leo."
				}, {
					"title": "Santé",
					"text": "Vivamus sagittis lacus vel augue laoreet rutrum faucibus."
				}, {
					"title": "Social & solidarités",
					"text": "Maecenas sed diam eget risus varius blandit sit amet."
				}, {
					"title": "Sports & loisirs",
					"text": "Cras mattis consectetur purus sit amet fermentum."
				}, {
					"title": "Urbanisme",
					"text": "Nullam quis risus eget urna mollis ornare vel eu leo."
				}, {
					"title": "Associations",
					"text": "Sed posuere consectetur est at lobortis."
				}, {
					"text": "Vous cherchez autre chose ?",
					"modifiers": ["more"]
				}]
			}
		},
		"jumbotron": {
			"image": "../images/modules/jumbotron/image.jpg",
			"title": "À la une",
			"text": "Revivez Nuit Blanche en photos et vidéo"
		},
		"news": {
			"heading": {
				"title": "En ce moment",
				"link": "Voir toutes les actualités"
			},
			"list": {
				"modifiers": ["news-cards"],
				"items": [{
					"href": "#",
					"image": "../images/modules/news-card/image.jpg",
					"category": "Mobilité",
					"title": "Découvrez les neuf Autolib' customisées qui vont sillonner Paris"
				}, {
					"href": "#",
					"image": "../images/modules/news-card/image.jpg",
					"category": "Culture et patrimoine",
					"title": "Le 1er étage de la Tour Eiffel se refait une jeunesse"
				}, {
					"href": "#",
					"image": "../images/modules/news-card/image.jpg",
					"category": "Environnement",
					"title": "Vers un plan parisien de lutte contre le gaspillage alimentaire"
				}]
			}
		},
		"que_faire": {
			"background": "../images/modules/que-faire/que-faire-back-img.jpg",
			"logo": {
				"href": "#",
				"image": "../images/modules/que-faire/que-faire-img.png"
			},
			"items": [{
				"href": "#",
				"text": "Que faire à Paris ce week-end ?"
			}, {
				"href": "#",
				"text": "Autour de la FIAC du 23 au 26 octobre"
			}, {
				"href": "#",
				"text": "Capitaine Futur et le voyage extraordinaire"
			}]
		},
		"get_involved": {
			"heading": {
				"title": "Participez !",
				"link": {
					"href": "#",
					"text": "Accéder à l’espace participatif"
				}
			},
			"items": [{
				"href": "#",
				"image": "../images/modules/get-involved/get-involved-img1.png",
				"text": "Votez pour le budget participatif"
			}, {
				"href": "#",
				"image": "../images/modules/get-involved/get-involved-img2.png",
				"text": "E-pétition, prenez la main"
			}]
		},
		"gallery": {
			"heading": {
				"title": "Vu par les Parisiens"
			},
			"pictures_gallery_ugc": {
				"title": "Hashtags de la semaine :",
				"hashtags": [{
					"url": "#",
					"text": "automneàparis"
				}, {
					"url": "#",
					"text": "nuit_blanche"
				}],
				"items": [{
					"image": "../images/modules/pictures-gallery-ugc/pictures-gallery-img1.jpg",
					"text": "@user"
				}, {
					"image": "../images/modules/pictures-gallery-ugc/pictures-gallery-img2.jpg",
					"text": "@user"
				}, {
					"image": "../images/modules/pictures-gallery-ugc/pictures-gallery-img3.jpg",
					"text": "@user"
				}, {
					"image": "../images/modules/pictures-gallery-ugc/pictures-gallery-img4.jpg",
					"text": "@user"
				}, {
					"image": "../images/modules/pictures-gallery-ugc/pictures-gallery-img5.jpg",
					"text": "@user"
				}, {
					"image": "../images/modules/pictures-gallery-ugc/pictures-gallery-img6.jpg",
					"text": "@user"
				}]
			}
		},
		"connected": {
			"heading": {
				"title": "Restez connecté",
				"modifiers": ["white"]
			},
			"newsletter": {
				"title": "L’essentiel de Paris",
				"text": "Chaque semaine, recevez l’essentiel de Paris pour savoir tout ce qu’il se passe dans la capitale.",
				"placeholder": "Entrez votre courriel",
				"inputId": "newsletter-input",
				"submit": "Valider",
				"otherNewsletters": {
					"text": "Autres newsletters et alertes",
					"url": "#"
				}
			},
			"jecoute": {
				"title": "@Parisjecoute",
				"items": [{
					"question": {
						"user": "@juliechanal",
						"time": "9h30",
						"text": "Bonjour, est-ce que la piscine Joséphine Baker sera ouverte cet été ?"
					},
					"answer": {
						"user": "@Parisjecoute",
						"time": "10h11",
						"text": "Oui, la piscine reste ouverte cet été."
					}
				}, {
					"question": {
						"user": "@JuliePetrucci",
						"time": "10h20",
						"text": "Bjr, je loue un local commercial. Quelles sont les démarches pour peindre la façade d'une autre couleur et y peindre le logo ?"
					},
					"answer": {
						"user": "@Parisjecoute",
						"time": "10h50",
						"text": "bjr, il faut déposer une déclaration préalable au 6 promenade Claude Lévi-Strauss dans le 13e http://t.co/h2uYPqkEOx"
					}
				}, {
					"question": {
						"user": "@billy141512",
						"time": "11h02",
						"text": "Bonjour, à propos du plan qualité air à Paris, les motos de 2012 (donc 3*) seront-elles autorisées en 2020 ? Pas clair."
					},
					"answer": {
						"user": "@Parisjecoute",
						"time": "11h14",
						"text": "bonjour, les motos de classe 3 étoiles seront bien concernées par ces mesures de restriction."
					}
				}],
				"button": {
					"href": "#",
					"text": "Posez votre question !"
				}
			},
			"social": {
				"title": "Restez connecté",
				"items": [{
					"icon": "twitter",
					"count": "219 000",
					"subscribe": {
						"url": "#",
						"text": "S’abonner à @paris"
					}
				}, {
					"icon": "facebook",
					"count": "2 508 576",
					"subscribe": {
						"url": "#",
						"text": "Suivre la page Facebook"
					}
				}, {
					"icon": "instagram",
					"count": "16 449",
					"subscribe": {
						"url": "#",
						"text": "S’abonner sur Instagram"
					}
				}]
			}
		},
		"municipal_team": {
			"heading": {
				"title": "La Maire et ses adjoints",
				"link": {
					"href": "#",
					"text": "L’Hôtel de ville"
				},
				"modifiers": ["white"]
			},
			"img": {
				"src": "../images/modules/municipal-team/image.jpg",
				"alt": "La Maire et ses adjoints"
			},
			"intro": "Maecenas sed diam eget risus varius blandit sit amet non magna.",
			"text": "Aenean eu leo quam. Pellentesque ornare sem lacinia quam venenatis vestibulum. Donec id elit non mi porta gravida at eget metus. Nullam id dolor id nibh ultricies vehicula ut id elit. Maecenas sed diam eget risus varius blandit sit amet non magna."
		},
		"header": {
			"title": "Paris",
			"h1": true,
			"languages": [{
				"abbr": "fr",
				"href": "/fr"
			}, {
				"abbr": "en",
				"href": "/en"
			}, {
				"abbr": "es",
				"href": "/es"
			}],
			"active_language": "fr",
			"search": {
				"visible": true
			},
			"marker": {
				"visible": true
			},
			"items": [{
				"href": "#",
				"text": "Services et infos pratiques"
			}, {
				"href": "#",
				"text": "L'Hôtel de ville"
			}, {
				"href": "#",
				"text": "Actualités"
			}],
			"participate": {
				"visible": true,
				"url": "#",
				"front": "Participez !",
				"back": "C'est à vous !"
			},
			"account": {
				"visible": true,
				"url": "#",
				"text": "Mon compte"
			}
		},
		"footer": {
			"logo": {
				"href": "#",
				"src": "../images/modules/footer/mairie-de-paris.png",
				"alt": "Logo de la Mairie de Paris"
			},
			"items": [{
				"href": "#",
				"text": "Visit Paris",
				"modifiers": ["footer-link-em"]
			}, {
				"href": "#",
				"text": "FAQ"
			}, {
				"href": "#",
				"text": "Newsletter"
			}, {
				"href": "#",
				"text": "Contact"
			}, {
				"href": "#",
				"text": "Presse"
			}, {
				"href": "#",
				"text": "Mentions légales"
			}, {
				"href": "#",
				"text": "Crédits"
			}, {
				"href": "#",
				"text": "Accessibilité"
			}]
		}
	}
};


/* GET home page. */
router.get('/', function(req, res, next) {
	res.render('templates/home/index', data);
});

module.exports = router;